hamilton/function_modifiers.py

```diff
@@ -626,71 +626,152 @@ def ensure_function_empty(fn: Callable):
 
 
 class does(function_modifiers_base.NodeCreator):
-    def __init__(self, replacing_function: Callable):
-        """
-        Constructor for a modifier that replaces the annotated functions functionality with something else.
+    def __init__(self, replacing_function: Callable, **argument_mapping: Union[str, List[str]]):
+        """Constructor for a modifier that replaces the annotated functions functionality with something else.
         Right now this has a very strict validation requirements to make compliance with the framework easy.
+        :param replacing_function: The function to replace the original function with
+        :param argument_mapping: A mapping of argument name in the replacing function to argument name in the decorating function
         """
         self.replacing_function = replacing_function
+        self.argument_mapping = argument_mapping
 
     @staticmethod
-    def ensure_output_types_match(fn: Callable, todo: Callable):
-        """
-        Ensures that the output types of two functions match.
+    def ensure_output_types_match(fn: Callable, replace_with: Callable):
```

Comment:
this would be clearer I think?
```suggestion
    def ensure_output_types_match(og_function: Callable, replacement_function: Callable):
```